encodings/sparse/src/lib.rs

```diff
@@ -107,6 +108,87 @@ impl SparseArray {
     pub fn fill_scalar(&self) -> &Scalar {
         &self.fill_value
     }
+
+    /// Encode given array as a SparseArray.
+    ///
+    /// Optionally provided fill value will be respected if the array is less than 90% null.
+    pub fn encode(array: &dyn Array, fill_value: Option<Scalar>) -> VortexResult<ArrayRef> {
+        let mask = array.validity_mask()?;
+
+        if mask.all_false() {
+            // Array is constant NULL
+            return Ok(
+                ConstantArray::new(Scalar::null(array.dtype().clone()), array.len()).into_array(),
```

Comment:
NullArray?

Reply:
NullArray is very special. For our usecase we almost always want ConstantArray with null scalar